### PR TITLE
Merge bugfix release into master

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Merge bugfix release v0.35.1 into master.

**Description of the Change:**
Fix how `qml.adjoint` is handled by `apply_lightning`.
Fix how diagonalizing gates are handled in measurements.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ pytest~=8.0.0
 pytest-benchmark
 pytest-cov
 pytest-mock
+pytest-xdist
```
Comment (AmintorDusko) [inline requirements-dev.txt:10]: Why are we adding this?

**Review by AmintorDusko — APPROVED** [review]
Thank you! Nice job here!

File: requirements-dev.txt (hunk 1 above)
Comment (vincentmr) [reply requirements-dev.txt:10]: It may have slipped there after switching branches. It will be used to accelerate native MCM tests in #621 .

**Review by maliasadi — APPROVED** [review]
:tada:

**Pull request merged**